test/hotspot/jtreg/runtime/cds/appcds/jigsaw/modulepath/AddOpens.java

```diff
@@ -78,29 +83,34 @@ public static void buildTestModule() throws Exception {
         // compile the modules and create the modular jar files
         buildTestModule();
-        String appClasses[] = {MAIN_CLASS};
-        // create an archive with both -cp and --module-path in the command line.
-        // Only the class in the modular jar in the --module-path will be archived;
-        // the class in the modular jar in the -cp won't be archived.
-        OutputAnalyzer output = TestCommon.createArchive(
-                                        destJar.toString(), appClasses,
-                                        "--module-path", moduleDir.toString(),
-                                        "-m", TEST_MODULE1);
-        TestCommon.checkDump(output);
-
-        // run with the archive using the same command line as in dump time
-        // plus the "--add-opens java.base/java.lang=com.simple" option.
-        // The main class should be loaded from the archive.
-        // The setaccessible(true) on the ClassLoader.defineClass method should
-        // be successful.
-        TestCommon.run( "-Xlog:class+load=trace",
-                        "-cp", destJar.toString(),
-                        "--add-opens", "java.base/java.lang=" + TEST_MODULE1,
-                        "--module-path", moduleDir.toString(),
-                        "-m", TEST_MODULE1, "with_add_opens")
-            .assertNormalExit(
-                "[class,load] com.simple.Main source: shared objects file",
-                "method.setAccessible succeeded!");
+        String appClasses[] = {MAIN_CLASS, "java/lang/ClassLoader"};
+
+        for (int i = 0; i < 2; i++) {
+            // create an archive with both -cp and --module-path, and with the
+            // --add-opens option if i == 1, in the command line.
+            // Only the class in the modular jar in the --module-path will be archived;
+            // the class in the modular jar in the -cp won't be archived.
+            OutputAnalyzer output = TestCommon.createArchive(
+                                            destJar.toString(), appClasses,
+                                            extraOpts[i][0], extraOpts[i][1],
+                                            "--module-path", moduleDir.toString(),
+                                            "-m", TEST_MODULE1);
+            TestCommon.checkDump(output);
+
+            // run with the archive using the same command line as in dump time
+            // plus the "--add-opens java.base/java.lang=com.simple" option.
+            // The main class should be loaded from the archive.
+            // The setaccessible(true) on the ClassLoader.defineClass method should
+            // be successful.
+            TestCommon.run( "-Xlog:class+load=trace",
+                            "-cp", destJar.toString(),
+                            "--add-opens", addOpensArg,
+                            "--module-path", moduleDir.toString(),
+                            "-m", TEST_MODULE1, "with_add_opens")
```

Comment:
Could you add a test case for `=ALL-UNNAMED` as well? That is frequently used by applications, and it takes a different code path inside the implementation.

I think this can be done by running `com.simple.Main` in the classpath rather than module path.

Reply:
Added the test case.